### PR TITLE
[Enhancement] (nereids)implement CreateCatalogCommand in nereids

### LINKED ISSUE #42594: [Enhancement] (nereids)implement CreateCatalogCommand in nereids
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

CreateCatalogCommand is supported in legacy planner but not in nereids(new doris fe planner). you need implement it in nereids.
The DorisParser.g4 has the syntax definition already(see bellow). And you can check CreateCatalogStmt.java in legacy planner to see what the command does and migrate it to nereids planner
``` unsupportedCreateStatement: CREATE CATALOG (IF NOT EXISTS)? catalogName=identifier (WITH RESOURCE resourceName=identifier)? (COMMENT STRING_LITERAL)? properties=propertyClause?```

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by msridhar78**
### What problem does this PR solve?

Issue Number: close #42594 

Related PR: #xxx

Problem Summary:
[Enhancement] (nereids)implement CreateCatalogCommand in nereids

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/CreateCatalogCommand.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Command to create a catalog in the Nereids planner.
+ */
+public class CreateCatalogCommand extends Command implements ForwardWithSync {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/CreateCatalogCommand.java:46]:
```suggestion
public class CreateCatalogCommand extends Command implements ForwardWithSync, NeedAuditEncryption {
```
implement NeedAuditEncryption interface

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/CreateCatalogCommand.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        properties.put(ExternalCatalog.CREATE_TIME, currentDateTime);
+        PropertyAnalyzer.checkCatalogProperties(properties, false);
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        Env.getCurrentEnv().getCatalogMgr().createCatalog(this);
+    }
+
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/CreateCatalogCommand.java:94]: implement these two methods from NeedAuditEncryption interface, take bellow as reference
```
@Override
public boolean needAuditEncryption() {
    return true
}

@Override
public String toSql() {
    StringBuilder stringBuilder = new StringBuilder();
    stringBuilder.append("CREATE CATALOG ").append("`").append(catalogName).append("`");
    if (!Strings.isNullOrEmpty(resource)) {
        stringBuilder.append(" WITH RESOURCE `").append(resource).append("`");
    }
    if (!Strings.isNullOrEmpty(comment)) {
        stringBuilder.append("\nCOMMENT \"").append(comment).append("\"");
    }
    if (properties.size() > 0) {
        stringBuilder.append("\nPROPERTIES (\n");
        stringBuilder.append(new PrintableMap<>(properties, "=", true, true, true));
        stringBuilder.append("\n)");
    }
    return stringBuilder.toString();
}
```

Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/CreateCatalogCommand.java:95]: done - please review

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run P0

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**